File: samples/snippets/fsharp/System/FuncT1,T2,T3,T4,TResult/Overview/Delegate.fs
module Delegate

// <Snippet1>
open System

type Searcher = delegate of (string * int * int * StringComparison) -> int

let title = "The House of the Seven Gables"
let finder = Searcher title.IndexOf
let mutable position = 0

while position > -1 do
    let characters = title.Length - position
    position <- 
        finder.Invoke("the", position, characters, StringComparison.InvariantCultureIgnoreCase)
    if position >= 0 then
        position <- position + 1
        printfn $"'The' found at position {position} in {title}."
// </Snippet1>

File: snippets/csharp/System/FuncT1,T2,T3,T4,TResult/Overview/Delegate.cs
// <Snippet1>
using System;

delegate int Searcher(string searchString, int start, int count,
                         StringComparison type);

public class NamedDelegateExample
{
    public static void Run()
    {
        string title = "The House of the Seven Gables";
        int position = 0;
        Searcher finder = title.IndexOf;
        do
        {
            int characters = title.Length - position;
            position = finder("the", position, characters,
                            StringComparison.InvariantCultureIgnoreCase);
            if (position >= 0)
            {
                position++;
                Console.WriteLine($"'The' found at position {position} in {title}.");
            }
        } while (position > 0);
    }
}
// </Snippet1>
